http-client/src/main/java/io/micronaut/http/client/DefaultHttpClient.java

```diff
@@ -1996,6 +1999,22 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
 
     }
 
+    private void setRedirectHeaders(@Nullable HttpRequest request, MutableHttpRequest<Object> redirectRequest) {
+        if (request != null) {
+            request.headers().forEach(header -> redirectRequest.header(header.getKey(), header.getValue()));
+        }
+    }
+
+    private void setRedirectHeaders(@Nullable io.micronaut.http.HttpRequest<?> request, MutableHttpRequest<Object> redirectRequest) {
+        if (request != null) {
+            final io.micronaut.http.HttpHeaders headers = request.getHeaders();
+            headers.names().forEach(name -> {
+                final Optional<String> headerValue = request.getHeaders().get(name, String.class);
+                headerValue.ifPresent(value -> redirectRequest.header(name, value));
+            });
```

Comment:
please `Iterator<Map.Entry<String, List<V>>> iterator()` method and simply call `toString()` on each value (if not null) to avoid the conversion cost here

Reply:
I am sorry to bother you with this, could you please verify if you are suggesting the following: 
```
 final Iterator<Map.Entry<String, List<String>>> headerIterator = request.getHeaders().iterator();
            while (headerIterator.hasNext()) {
                final Map.Entry<String, List<String>> nextHeader = headerIterator.next();
                if (nextHeader.getValue() != null) {
                    redirectRequest.header(nextHeader.getKey(), nextHeader.getValue().toString());
                }
            }
```

Above code does not work where value is a List of String as `nextHeader.getValue().toString()` would convert it to `[text/plain,application/json]`

Reply:
You have to iterate over the list